graphql/schemabuilder/schema.go

```diff
@@ -199,6 +199,50 @@ func (s *Schema) Mutation() *Object {
 	return s.Object("Mutation", mutation{})
 }
 
+const DuplicateTypeNameErrFormat string = "%s type name is duplicated in packages %s and %s"
+
+func checkTypeNameUniqueness(typ reflect.Type, typesToPackages map[string]string) error {
+	// Invoke typ.Elem() until we get a struct or scalar type.
+	for kind := typ.Kind(); kind == reflect.Slice || kind == reflect.Ptr || kind == reflect.Map; kind = typ.Kind() {
+		typ = typ.Elem()
+	}
+
+	// Duplicate scalar type alias names are ok because scalar type aliases
+	// are converted to the underlying type.
+	if _, ok := getScalar(typ); ok {
+		return nil
+	}
+	// The union type marker is a special type to denote union types and is not
+	// included in the introspection result.
+	if typ == unionType {
+		return nil
+	}
+
+	packageName := typ.PkgPath()
+	typeName := typ.Name()
+
+	if seenPackage, ok := typesToPackages[typeName]; ok && seenPackage != packageName {
+		return fmt.Errorf(DuplicateTypeNameErrFormat, typeName, seenPackage, packageName)
+	} else if ok && seenPackage == packageName {
+		return nil
+	}
+
+	typesToPackages[typeName] = packageName
+
+	if typ.Kind() != reflect.Struct {
```

Comment:
How can we run into this?

Reply:
Ah this was because of the special `graphql.SelectionSet` type. This type is actually excluded from the introspection result so I decided to do a type check for it instead here: https://github.com/samsarahq/thunder/pull/510/commits/423b78717b35e964c018fccb076298255eb2a94e#diff-414e6a761a1e5643cfd90f7d667ca5d2d5037726aee49a5864c7ace2a8b7b453R290